### PR TITLE
Common partition tables for Espressif boards

### PR DESCRIPTION
**Pull request by marekmatej**
This PR presents generic purpose partition tables to prevent proliferating the board `dts`
with the flash partition table.

### CODE REVIEW

File: dts/common/espressif/partitions_default_8M.dtsi [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		};
+
+		/* Slot1: 3904kB */
+		slot1_partition: partition@410000 {
+			label = "image-1";
+			reg = <0x410000 0x3d0000>;
+		};
+
+		/* Optional scratch */
+		scratch_partition: partition@7e0000 {
```
Comment (sylvioalves) [inline dts/common/espressif/partitions_default_8M.dtsi:39]: I think scratch partition could be removed from everywhere.

Comment (marekmatej) [reply dts/common/espressif/partitions_default_8M.dtsi]: I know it is not used anymore in MCUboot swap, but there are a few samples that still use the `scratch_partition` so I would rather leave it there.

**Comment by marekmatej** [comment]
CI failures are fixed by the https://github.com/mcu-tools/mcuboot/pull/2040 
Merge to https://github.com/zephyrproject-rtos/mcuboot is pending.

**Review by uLipe — DISMISSED** [review]
LGTM so far, question: why the 0x1000 appended at the partition name file?

**Comment by marekmatej** [comment]
> LGTM so far, question: why the 0x1000 appended at the partition name file?

That is the boot offset for some Espressif devices (ESP32, ESP32-s2, ... more to come).
And I'm also thinking of adding the boot offset to all partition files 4ex. `0x0`, seems handy. What do you think?

**Comment by uLipe** [comment]
> > LGTM so far, question: why the 0x1000 appended at the partition name file?
> 
> That is the boot offset for some Espressif devices (ESP32, ESP32-s2, ... more to come). And I'm also thinking of adding the boot offset to all partition files 4ex. `0x0`, seems handy. What do you think?

I like the idea since it is documented somewhere for reference.

**Review by wmrsouza — DISMISSED** [review]

**Review by LucasTambor — DISMISSED** [review]

**Comment by marekmatej** [comment]
@uLipe PTAL again

**Review by uLipe — DISMISSED** [review]

**Review by uLipe — APPROVED** [review]

**Review by wmrsouza — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**